lib/appoptics_apm/xtrace.rb

```diff
@@ -33,6 +33,19 @@ def sampled?(xtrace)
         valid?(xtrace) && xtrace[59].to_i & 1 == 1
```

Comment:
interesting. in node i would just check if last char was a '1'. your approach is more future proof (if more flags come along) and might be faster in ruby. but node's immutable strings make it an address compare. i wouldn't have thought about this approach at all. (i do have a low level c function that checks the bit of a non-string xtrace where i leave it most of the time.)

Reply:
I learned this from Daniel's C code ;)